### PR TITLE
Support XDG_CONFIG_HOME dir for pwn.conf

### PR DESCRIPTION
**Pull request by WGH-**
~/.pwn.conf support is not removed.

### CODE REVIEW

**Comment by heapcrash** [comment]
I don't mind having this as an ADDITIONAL configuration that gets loaded, but we should keep `~/.pwn.conf` the canonical one (and the one recommended by documentation).

File: pwnlib/update.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
     $ echo never > ~/.cache/.pwntools-cache-*/update
 
-Or adding the following lines to ~/.pwn.conf (or system-wide /etc/pwn.conf):
+Or adding the following lines to ~/.config/pwn.conf (or system-wide /etc/pwn.conf):
```
Comment (heapcrash) [inline pwnlib/update.py]: Don't change the default

File: pwnlib/update.py [hunk 2]
```diff
@@ -191,10 +191,13 @@ def perform_check(prerelease=current_version.is_prerelease):
     return command
 
 def check_automatically():
+    xdg_config_home = os.environ.get('XDG_CONFIG_HOME') or "~/.config"
```
Comment (heapcrash) [inline pwnlib/update.py:194]: I would recommend loading BOTH files, not one vs. the other.

File: pwnlib/update.py [hunk 3]
```diff
@@ -191,10 +191,13 @@ def perform_check(prerelease=current_version.is_prerelease):
...
+
     if should_check():
         message  = ["Checking for new versions of %s" % package_name]
         message += ["To disable this functionality, set the contents of %s to 'never' (old way)." % cache_file()]
-        message += ["""Or add the following lines to ~/.pwn.conf (or /etc/pwn.conf system-wide):
+        message += ["Or add the following lines to %s (or /etc/pwn.conf system-wide):" % (xdg_config_home + "/pwn.conf")]
```
Comment (heapcrash) [inline pwnlib/update.py]: List both locations, not one or the other.

Comment (heapcrash) [reply pwnlib/update.py]: Also we need to handle the case where XDG_CONFIG_HOME is not set (e.g. macOS) and not recommend that path.

**Review by heapcrash — CHANGES_REQUESTED** [review]

File: pwnlib/update.py (hunk 2 above)
Comment (WGH-) [reply pwnlib/update.py:194]: All programs I've ever seen use the same logic I suggested, e.g. 

https://github.com/git/git/blob/e6362826a0409539642a5738db61827e5978e2e4/Documentation/git-config.txt#L293-L294

https://github.com/htop-dev/htop/blob/12c2337939cbdee21c437c28646f128667a162de/Settings.c#L392-L403

Also, the default value for `$XDG_CONFIG_HOME` is defined to be `~/.config`. In fact, I've never seen a system with explicitly defined `$XDG_CONFIG_HOME` environment variable yet.

Comment (WGH-) [reply pwnlib/update.py:194]: pwntools already does it like this for `$XDG_CACHE_HOME`, see

https://github.com/Gallopsled/pwntools/blob/082de2cc7e6e1dce00d00851fae14741038778f0/pwnlib/context/__init__.py#L1236-L1237

File: pwnlib/update.py (hunk 3 above)
Comment (WGH-) [reply pwnlib/update.py]: Are you referring to `~/.config` vs `$XDG_CONFIG_HOME` discussion above, or you're suggesting that all three of `/etc/pwn.conf`, `~/.config/pwn.conf` and `~/.pwn.conf` should be mentioned here?

Comment (heapcrash) [reply pwnlib/update.py]: > all three of /etc/pwn.conf, ~/.config/pwn.conf and ~/.pwn.conf should be mentioned here

This, we should continue to support loading from ~/.pwn.conf even if $XDG_HOME/pwn.conf exists.  If they both exist, load them both, don't really care about ordering.

Comment (heapcrash) [reply pwnlib/update.py]: Probably ordering should be

/etc/pwn.conf
XDG_CONFIG_HOME/pwn.conf
~/.pwn.conf

This keeps the current behavior (~/.pwn.conf wins), while adding support for the XDG_CONFIG_HOME you're proposing.

Comment (WGH-) [reply pwnlib/update.py]: I suppose the macOS question was answered in the duscussion above?

**Review by Arusekk — APPROVED** [review]

File: pwnlib/update.py (hunk 3 above)
Comment (heapcrash) [reply pwnlib/update.py]: Yeah, if `$XDG_CONFIG_HOME` is not set, then just don't try to load that file.

**Comment by heapcrash** [comment]
Fixed CHANGELOG, merging

**Pull request merged**